clang/include/clang/CIR/Dialect/IR/CIROps.td

```diff
@@ -3853,6 +3853,36 @@ def MemCpyOp : CIR_Op<"libc.memcpy"> {
     mlir::cir::IntType getLenTy() { return getLen().getType(); }
   }];
 }
+//===----------------------------------------------------------------------===//
+// MemSetOp
+//===----------------------------------------------------------------------===//
+
+def MemSetOp : CIR_Op<"libc.memset"> {
+  let arguments = (ins Arg<VoidPtr, "", [MemWrite]>:$dst,
+                       UInt8:$val,
+                       PrimitiveUInt:$len);
+  let summary = "Equivalent to libc's `memset`";
+  let description = [{
+    Given the CIR pointer, `dst`, `cir.libc.memset` will set the first `len`
+    bytes of the memory pointed by `dst` to the specified `val`.
+
+    Examples:
+
+    ```mlir
+      // Set 2 bytes from a struct to 0:
+      %2 = cir.const #cir.int<2> : !u32i
+      %3 = cir.const #cir.int<0> : !u32i
+      %zero = cir.cast(integral, %3 : !s32i), !u8i
+      cir.libc.memset %2 bytes from %struct set to %zero : !u64i, !u8i -> !cir.ptr<!void>
+    ```
+  }];
+
+  let assemblyFormat = [{
+    $len `bytes` `from` $dst `set` `to` $val attr-dict
+    `:` type($len) `,` type($val) `->` qualified(type($dst))
```

Comment:
By convention we use the `->` syntax to separate parameters and results. Since this opeartion does not have any results, just list parameter types in order is sufficient:

```sugggestion
    `:` qualified(type($dst)) `,` type($val) `,` type($len)
```